### ISSUE TITLE
Google Maps plugin: native interface incomplete

### ISSUE THREAD

**Issue by duzenko** [flutter/flutter#77712]
File: `packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

Function:  `boolean onMarkerTap(String googleMarkerId) `

It always returns `false` and does not allow the dart code to control if it wants to consume the event and suppress the default behavior.

## Steps to Reproduce

1. Have a `GoogleMap` widget in your app with some markers.
2. Tap on any marker.

**Expected results:** The dart side should have control over the default behavior.

**Actual results:** The default behavior (map scrolls to marker) always occurs.

**Comment by darshankawar** [comment]
@duzenko 
Can you elaborate on the expected result ? `The dart side should have control over the default behavior.` Usually when you tap on the marker, it is expected to scroll towards it.

**Comment by duzenko** [comment]
> 
> 
> @duzenko
> Can you elaborate on the expected result ? `The dart side should have control over the default behavior.` Usually when you tap on the marker, it is expected to scroll towards it.

Sometimes we need the map not to scroll.

**Comment by darshankawar** [comment]
@duzenko 
So you are looking for a way /  property wherein you can disable the scroll to marker behavior ?

**Comment by duzenko** [comment]
> 
> 
> @duzenko
> So you are looking for a way / property wherein you can disable the scroll to marker behavior ?

No, I want the flutter marker tap interface to be inline with with the native interface.

**Comment by darshankawar** [comment]
@duzenko 
Sorry if I am asking this question again, but can you provide more context on what native interface you are looking for ?

**Comment by duzenko** [comment]
> 
> 
> @duzenko
> Sorry if I am asking this question again, but can you provide more context on what native interface you are looking for ?

`onMarkerClick(Marker                   marker)`

https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap.OnMarkerClickListener#onMarkerClick(com.google.android.gms.maps.model.Marker)

**Comment by Kurogoma4D** [comment]
I want this.
I used google_maps_flutter package as just display map between two points, and it didn't need to add any interactions.

**Comment by stuartmorgan-g** [comment]
This is blocked on both thread merging and at least a partial migration to FFI, since the proposal requires a fully synchronous native->Dart->native flow.
